src/js/keyboardInput.js

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2015 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+(function () {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.firstDiscovery");
+
+    gpii.firstDiscovery.charFromKeypress = function (e) {
+        return e.which ? String.fromCharCode(e.which) : "";
+    };
+
+    fluid.defaults("gpii.firstDiscovery.usKeymap", {
+        gradeNames: ["fluid.littleComponent", "autoInit"],
+        invokers: {
+            isShiftEvent: {
+                funcName: "gpii.firstDiscovery.usKeymap.isShiftEvent",
+                args: ["{that}", "{arguments}.0"]
+            },
+            isLowerCaseLetter: {
+                funcName: "gpii.firstDiscovery.usKeymap.isLowerCaseLetter",
+                args: ["{arguments}.0"]
+            },
+            canShiftChar: {
+                funcName: "gpii.firstDiscovery.usKeymap.canShiftChar",
+                args: ["{that}.shiftTable", "{arguments}.0"]
+            },
+            getShiftedChar: {
+                funcName: "gpii.firstDiscovery.usKeymap.getShiftedChar",
+                args: ["{that}.shiftTable", "{arguments}.0"]
+            }
+        },
+        members: {
+            shiftKeyCode: 16,
+            shiftTable: {
+                "`": "~",
+                "1": "!",
+                "2": "@",
+                "3": "#",
+                "4": "$",
+                "5": "%",
+                "6": "^",
+                "7": "&",
+                "8": "*",
+                "9": "(",
+                "0": ")",
+                "-": "_",
+                "=": "+",
+                "[": "{",
+                "]": "}",
+                "\\": "|",
+                ";": ":",
+                "'": "\"",
+                ",": "<",
+                ".": ">",
+                "/": "?"
+            }
+        }
+    });
+
+    gpii.firstDiscovery.usKeymap.isShiftEvent = function (keymap, e) {
+        return e.which === keymap.shiftKeyCode;
+    };
+
+    gpii.firstDiscovery.usKeymap.isLowerCaseLetter = function (ch) {
+        var charCode = ch.charCodeAt(0);
+        return charCode >= 97 && charCode <= 122;
```

Comment:
I decided to switch to using named values rather than number literals "keymap.charCodeLowerCaseA" and "keymap.charCodeLowerCaseZ"